ivy/functional/frontends/paddle/tensor/math.py

```diff
@@ -487,6 +487,17 @@ def subtract(x, y, name=None):
     return ivy.subtract(x, y)
 
 
+@with_supported_dtypes(
+    {"2.5.1 and below": ("bool", "int32", "int64", "float32", "float64")}, "paddle"
+)
+@to_ivy_arrays_and_back
+def sum(x, axis=None, keepdim=False, name=None):
+    if x.dtype == ivy.int32:
```

Comment:
What is the reason for doing this?

Reply:
@yopknopixx the paddle documentation says if the dtype of input array is int32 the output dtype should be int64. ivy.sum does not cover this edge case and if this dtype conversion is not done all tests fail with "expected dtype int64 got int32". to address this issue this is done. Thanks

Reply:
@yopknopixx I rechecked and found that even when input array dtype is bool paddle.sum converts it to int64, I have added condition for that too. and ivy.sum documentation says whenever such expected dtype is different from dtype of array we have to convert the dtype to desired type before passing to the function. so added code to do that too.